src/properties.rs

```diff
@@ -27,7 +27,7 @@ impl FromStr for Color {
             s
         };
         match s.len() {
-            6 => {
+            6 if s.is_ascii() => {
```

Comment:
Does this really suffice? The docs for `from_str_radix` say "This function panics if radix is not in the range from 2 to 36.". I think we want [`is_ascii_hexdigit`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_hexdigit)?

Reply:
The radix in this case is the constant value `16` and `from_str_radix` has its own graceful error handling for invalid inputs (first arg)

Reply:
Oh alright, I was misunderstanding why this panicked. Apparently it's about invalid UTF-8?

Reply:
Yeah, `String`s and string slices (`str`) must be valid UTF-8 encoded strings. Iterating over the characters of a string in Rust is typically done with an `s.chars()` iterator.

Reply:
Alright, thanks for the explanation! So indeed a check on ascii suffices here.